staging-service-shared/src/main/java/com/github/cafdataprocessing/services/staging/utils/BatchProgressTracker.java

```diff
@@ -0,0 +1,72 @@
+/*
+ * Copyright 2019-2022 Micro Focus or one of its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.github.cafdataprocessing.services.staging.utils;
+
+import org.apache.commons.fileupload.ProgressListener;
+import org.apache.commons.fileupload.servlet.ServletFileUpload;
+
+import java.time.Instant;
+import java.time.temporal.ChronoUnit;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.ConcurrentMap;
+
+public class BatchProgressTracker
+{
+    private BatchProgressTracker()
+    {
+        throw new IllegalStateException();
+    }
+
+    public static ConcurrentMap<String, Tracker> getInProgressTrackerMap()
+    {
+        return inProgressTrackerMap;
+    }
+
+    private static final ConcurrentHashMap<String, Tracker> inProgressTrackerMap = new ConcurrentHashMap<>();
+
+    public static void updateTracker(ServletFileUpload fileUpload)
+    {
+        fileUpload.setProgressListener(createProgressListener());
+    }
+
+    private static ProgressListener createProgressListener()
+    {
+        Tracker tracker = new Tracker();
+        tracker.setUploadStartTime(Instant.now());
+        return new ProgressListener()
+        {
+            private long megaBytes = -1;
+            @Override
+            public void update(long totalBytesRead, long contentLength, int item) {
+                long totalBytesInMBytes = totalBytesRead / 1000000;
```

Comment:
Use underscores when you have big numbers to make them easier to read:

```suggestion
                long totalBytesInMBytes = totalBytesRead / 1_000_000;
```